airbyte-integrations/connectors/source-freshsales/source_freshsales/source.py

```diff
@@ -202,34 +177,26 @@ class UpcomingAppointments(FreshsalesStream):
     object_name = "appointments"
     filter_value = "upcoming"
 
-    def path(
-        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
-    ) -> str:
-        return f"{self.object_name}"
-
-    def request_params(
-        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
-    ) -> MutableMapping[str, Any]:
-        params = super().request_params(stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
-        params["filter"] = self.filter_value
-        return params
-
 
 # Source
 class SourceFreshsales(AbstractSource):
+    @staticmethod
+    def get_input_stream_args(api_key: str, domain_name: str) -> Mapping[str, Any]:
+        return {
+            "authenticator": TokenAuthenticator(token=api_key, auth_method="Token"),
+            "domain_name": domain_name,
+        }
+
     def check_connection(self, logger, config) -> Tuple[bool, any]:
-        auth = TokenAuthenticator(token=f'token={config["api_key"]}', auth_method="Token").get_auth_header()
-        url = f'https://{config["domain_name"]}/crm/sales/api/contacts/filters'
+        stream = Contacts(**self.get_input_stream_args(config["api_key"], config["domain_name"]))
         try:
-            session = requests.get(url, headers=auth)
-            session.raise_for_status()
+            next(stream.read_records(sync_mode=None))
             return True, None
         except requests.exceptions.RequestException as e:
             return False, e
```

Comment:
I think we have tooling in the CDK for this check, which handles edge cases a bit better - something like
```
check_stream = CheckStream(stream_names=["contacts"], parameters={})
return check_stream.check_connection(self, logger, config)
```

Reply:
I've checked the Python version of `airbyte-cdk` once again and didn't find anything related to `CheckStream()` class or anything related to it. But we do have such for `declarative` source (Low-Code). So, leaving it as is.

Reply:
I prefer not to mix parts since lots of other dependencies could bring more inconsistencies with future updates of the CDK, if you don't mind.